hedera-mirror-importer/src/main/resources/db/migration/v1/V1.46.3__smart_contracts.sql

```diff
@@ -0,0 +1,113 @@
+-- Enhances smart contract storage to include all protobuf fields and to track changes to it over time
+
+-- contract
+create table if not exists contract
+(
+    like entity,
+    file_id     bigint null,
+    obtainer_id bigint null
+);
+
+alter table if exists contract
+    add primary key (id);
+
+-- Ensure entities associated with contract transactions are properly marked as contracts
+with contract_transaction as (
```

Comment:
why update contract entities only to then delete them.
Could we skip the entity update and do it in one go like below?

```suggestion
-- get contract entityIds from contract modifying transactions, then delete from entity table and insert deleted rows into contract
with contract_transactions as (
    select distinct entity_id
    from transaction
    where entity_id is not null
      and type in (7, 8, 9, 22)
),
contract_entities as (
    delete from only entity e where e.id = contract_transactions.entity_id returning *
)
insert
into contract
select auto_renew_account_id, auto_renew_period, created_timestamp, deleted, expiration_timestamp, id, key, memo, num, public_key, proxy_account_id, realm, shard, type, timestamp_range
from contract_entities;
```

Reply:
Agreed, better safe than sorry.
I'm okay with the separate steps